extensions/sql-bindings/src/test/common/azureFunctionsUtils.test.ts

```diff
@@ -236,6 +239,117 @@ describe('AzureFunctionUtils', function (): void {
 		});
 	});
 
+	describe('Get Azure Function Files', function (): void {
+		it('Should return undefined if not azure function projects are found', async () => {
+			// set workspace folder for testing
+			sinon.replaceGetter(vscode.workspace, 'workspaceFolders', () => {
+				return <vscode.WorkspaceFolder[]>[{
+					uri: {
+						fsPath: '/temp/'
+					},
+				}];
+			});
+			let result = await azureFunctionsUtils.getAzureFunctionProject();
```

Comment:
This one is supposed to return undefined in the scenario we do not find any files / azure function projects. For the next tests I need to stub out the findFiles call as if I don't then it would not go return the expected result from the `getAzureFunctionProjectFiles` as well as `getHostFiles` function calls.

Reply:
Yes, but what I'm saying is that the call is doing a findFiles call right? And just happening to return undefined because the path doesn't exist on disk? You should avoid doing that if so - instead mocking out findFiles to return undefined/empty as appropriate for your scenario. Otherwise you risk IO issues or random stuff like the file just happening to exist on disk messing up the test.